server/client.go

```diff
@@ -145,11 +159,41 @@ func (cs *clientStore) isValidWithTimeout(ID string, connID []byte, timeout time
 		// We timed out, remove the entry in the map
 		cs.Lock()
 		delete(cs.waitOnRegister, ID)
+		cs.addToKnownInvalid(ID, connID)
 		cs.Unlock()
 		return false
 	}
 }
 
+func (cs *clientStore) addToKnownInvalid(ID string, connID []byte) {
+	cID := string(connID)
```

Comment:
Since you do this twice maybe make it a small func, would simplify knownToBeInvalid as well.

Reply:
I actually moved to using clientID+":"+connID as the key of the map. It simplifies the code I think. If you could have a second look. Thanks!